diff -Naur tslib-1.0/configure.ac tslib-1.0-modded/configure.ac
--- tslib-1.0/configure.ac	2006-08-16 17:48:28.000000000 -0400
+++ tslib-1.0-modded/configure.ac	2017-05-08 15:16:23.959701411 -0400
@@ -43,7 +43,6 @@
 
 # Checks for library functions.
 AC_PROG_GCC_TRADITIONAL
-AC_FUNC_MALLOC
 AC_FUNC_MMAP
 AC_FUNC_SELECT_ARGTYPES
 AC_TYPE_SIGNAL
diff -Naur tslib-1.0/tests/fbutils.c tslib-1.0-modded/tests/fbutils.c
--- tslib-1.0/tests/fbutils.c	2006-08-16 17:48:28.000000000 -0400
+++ tslib-1.0-modded/tests/fbutils.c	2017-05-08 15:16:55.191702003 -0400
@@ -193,18 +193,21 @@
 {
 	int i,j,bits;
 
-	for (i = 0; i < font_vga_8x8.height; i++) {
+	for (i = font_vga_8x8.height - 1; i >= 0; i--) {
 		bits = font_vga_8x8.data [font_vga_8x8.height * c + i];
-		for (j = 0; j < font_vga_8x8.width; j++, bits <<= 1)
+		for (j = font_vga_8x8.width - 1; j >= 0; j--, bits <<= 1)
 			if (bits & 0x80)
-				pixel (x + j, y + i, colidx);
+				pixel (x + j - font_vga_8x8.width, y - i + font_vga_8x8.height, colidx);
 	}
 }
 
 void put_string(int x, int y, char *s, unsigned colidx)
 {
 	int i;
-	for (i = 0; *s; i++, x += font_vga_8x8.width, s++)
+	int length;
+	length = strlen(s);
+	x += length * font_vga_8x8.width;
+	for (i = 0; *s; i++, x -= font_vga_8x8.width, s++)
 		put_char (x, y, *s, colidx);
 }
 
diff -Naur tslib-1.0/tests/ts_calibrate.c tslib-1.0-modded/tests/ts_calibrate.c
--- tslib-1.0/tests/ts_calibrate.c	2006-08-16 17:48:28.000000000 -0400
+++ tslib-1.0-modded/tests/ts_calibrate.c	2017-05-08 15:17:04.663702183 -0400
@@ -204,8 +204,8 @@
 		setcolor (i, palette [i]);
 
 	put_string_center (xres / 2, yres / 4,
-			   "TSLIB calibration utility", 1);
-	put_string_center (xres / 2, yres / 4 + 20,
+			   "Touchscreen calibration utility", 1);
+	put_string_center (xres / 2, yres / 4 - 20,
 			   "Touch crosshair to calibrate", 2);
 
 	printf("xres = %d, yres = %d\n", xres, yres);
